bundles/framework/hierarchical-layerlist/view/LayerGroupTab.js

```diff
@@ -1058,6 +1071,25 @@ Oskari.clazz.define(
                     me._toggleLayerCheckboxes(layer.getId(), true);
                 });
             });
+
+            me.getJsTreeElement().on('redraw.jstree',function(nodes){
+                me._updateAllTools();
+                me._updateLayerCountsAndGroupsVisibility(true);
+            });
+        },
+
+        /**
+         * Set no result message visible
+         * @method setNoResultMessageVisible
+         * @param  {Boolean}                  visible has message visible
+         */
+        setNoResultMessageVisible: function(visible) {
+            var noResultsElement = jQuery('.hierarchical-layerlist-search-noresults');
```

Comment:
Same here, don't search the whole document. Just from where you expect to find it.